### PR TITLE
feat: implement stream density python

### PR DESCRIPTION
**Pull request by jim-wang-intel**
CLOSES: #42

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/performance-tools/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [x] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**42**

## Test Instructions if applicable

- git clone this PR
- for unit tests: run `make python-test`
- for smoke tests: run `python benchmark.py --retail_use_case_root ../../retail-use-cases   --target_fps 12.1 --density_increment 1   --init_duration 40   --compose_file ../../retail-use-cases/use-cases/grpc_python/docker-compose_grpc_python.yml`

### CODE REVIEW

**Review by antoniomtz — CHANGES_REQUESTED** [review]

File: benchmark-scripts/test_stream_density_results/pipeline20240405141414071831277_gst.log [hunk 1]
```diff
@@ -0,0 +1,42 @@
+39.79
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply benchmark-scripts/test_stream_density_results/pipeline20240405141414071831277_gst.log:1]: no, those are test files for unit testing

File: benchmark-scripts/test_stream_density_results/pipeline20240405141713198120407_gst.log [hunk 2]
```diff
@@ -0,0 +1,106 @@
+41.41
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply benchmark-scripts/test_stream_density_results/pipeline20240405141713198120407_gst.log:1]: ditto

File: benchmark-scripts/test_stream_density_results/r20240405141414071831277_gst.jsonl [hunk 3]
```diff
@@ -0,0 +1,183 @@
+{"resolution":{"height":1080,"width":1920},"timestamp":0}
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply benchmark-scripts/test_stream_density_results/r20240405141414071831277_gst.jsonl:1]: ditto

File: benchmark-scripts/test_stream_density_results/pipeline20240405141414071831277_gst.log (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoniomtz) [reply benchmark-scripts/test_stream_density_results/pipeline20240405141414071831277_gst.log:1]: ok got it

**Review by antoniomtz — APPROVED** [review]
LGMT. tested it and ran successfully

**Pull request merged**